scripts/update_changelog.py

```diff
@@ -0,0 +1,46 @@
+#-*- coding: utf-8-unix -*-
+import re
+import subprocess
+from datetime import datetime
+
+# Function to extract version from versions.gradle
+def get_version_from_gradle():
+    with open("versions.gradle", "r") as f:
+        for line in f:
+            if "version" in line:
+                match = re.search(r"version\s*=\s*['\"]([^'\"]+)['\"]", line)
+                if match:
+                    return match.group(1)
+    raise ValueError("Version not found in versions.gradle")
+
+# Get the current version
+version = get_version_from_gradle()
+
+# Get the latest commit message
+commit_msg = subprocess.getoutput("git log -1 --pretty=%B").strip().replace("\\r\\n", '\n')
+
+# Get the commit date in YYYY-MM-DD format
+commit_date_raw = subprocess.getoutput("git log -1 --pretty=%cd --date=short")
+commit_date = datetime.strptime(commit_date_raw, "%Y-%m-%d").strftime("%Y-%m-%d")
+
+# Read the current changelog (or initialize it if it doesn’t exist)
+try:
+    with open("CHANGELOG.md", "r") as f:
+        changelog = f.read()
+except FileNotFoundError:
+    changelog = "# Changelog\n"
+
+# Find or create the version section
+version_section = f"## [{version}] - {commit_date}"
+if version_section not in changelog:
+    changelog = changelog.replace("This is the changelog for *casual java* and all changes are listed in this document.\n", f"This is the changelog for *casual java* and all changes are listed in this document.\n\n{version_section}\n", 1)
```

Comment:
Use a variable for the string? Potentially have it shared between all three locations including the create-initial-changelog.sh - wait - why is that a bash, but this is now python. Do we need python / both?!